webpack/components/extensions/HostDetails/Tabs/ModuleStreamsTab/ModuleStreamsTab.js

```diff
@@ -0,0 +1,200 @@
+  } = useBulkSelect({
+    results,
+    metadata,
+    isSelectable: result => false,
+  });
+
+  if (!hostId) return <Skeleton />;
+
+  const rowActions = [
+    {
+      title: __('Install'), disabled: true,
+    },
+    {
+      title: __('Update'), disabled: true,
+    },
+    {
+      title: __('Enable'), disabled: true,
+    },
+    {
+      title: __('Disable'), disabled: true,
+    },
+    {
+      title: __('Reset'), disabled: true,
+    },
+    {
+      title: __('Remove'), disabled: true,
+    },
+  ];
+
+  const EnabledIcon = ({ stream, streamStatus }) => {
+    switch (true) {
+      case (stream === 'Disabled' && streamStatus === 'Not installed'):
+        return <InactiveText text={streamStatus} />;
+      case (stream === 'Default' && streamStatus === 'Not installed'):
+        return <InactiveText text={streamStatus} />;
+      case (stream === 'Enabled' && streamStatus === 'Not installed'):
+        return <TableText wrapModifier="nowrap"> {streamStatus} </TableText>;
+      case (stream === 'Enabled' && streamStatus === 'Up-to-date'):
+        return <><TableText wrapModifier="nowrap"> </TableText><CheckIcon color="green" /> {streamStatus}</>;
+      case (stream === 'Enabled' && streamStatus === 'Upgradable'):
+        return <LongArrowAltUpIcon color="blue"> {streamStatus}</LongArrowAltUpIcon>;
+      default:
+        return null;
+    }
+  };
+
+  EnabledIcon.propTypes = {
+    stream: PropTypes.string.isRequired,
+    streamStatus: PropTypes.string.isRequired,
+  };
+
+  return (
+    <div>
+      <div id="modulestreams-tab">
+        <TableWrapper
+          {...{
+          metadata,
+          emptyContentTitle,
+          emptyContentBody,
+          emptySearchTitle,
+          emptySearchBody,
+          searchQuery,
+          updateSearchQuery,
+          fetchItems,
+          status,
+          rowActions,
+        }}
+          additionalListeners={[hostId]}
+          fetchItems={fetchItems}
+          autocompleteEndpoint={`/hosts/${hostId}/module_streams/auto_complete_search`}
+          foremanApiAutoComplete
+          rowsCount={results?.length}
+          variant={TableVariant.compact}
+        >
+          <Thead>
+            <Tr>
+              {columnHeaders.map(col =>
+                <Th key={col}>{col}</Th>)}
+            </Tr>
+          </Thead>
+          <Tbody>
+            {results?.map((moduleStream, index) => {
+              const {
+                id,
+                status: moduleStreamStatus,
+                name,
+                stream,
+                installed_profiles: installedProfiles,
+                install_status: installStatus,
+                module_spec: moduleSpec,
+              } = moduleStream;
+
+              let color;
+              let labelVariant;
+              let disabledText = false;
+              /* eslint-disable max-len */
+              let streamStatus = moduleStreamStatus.charAt(0).toUpperCase() + moduleStreamStatus.slice(1);
```

Comment:
I would also safe these types of values in case of "" or undefined. 
moduleStreamStatus?.charAt(0)?.etc...